rt_entt_codegen/core/print_sys_exec.cc

```diff
@@ -578,392 +273,183 @@ static auto print_apply_pendings(
-	ecsact::codegen_plugin_context&                            ctx,
-	const ecsact::rt_entt_codegen::ecsact_entt_details&        details,
-	const ecsact::rt_entt_codegen::ecsact_entt_system_details& sys_details,
-	const ecsact::rt_entt_codegen::core::print_execute_systems_options options
-) -> void {
-	using ecsact::cc_lang_support::cpp_identifier;
-	using ecsact::cpp_codegen_plugin_util::block;
-	using ecsact::meta::decl_full_name;
-	using ecsact::rt_entt_codegen::ecsact_entt_system_details;
-	using ecsact::rt_entt_codegen::system_util::create_context_struct_name;
-	using ecsact::rt_entt_codegen::system_util::create_context_var_name;
+static auto setup_system_providers(system_like_id_variant sys_like_id_variant
+) -> std::vector<std::shared_ptr<system_provider>> {
+	using ecsact::rt_entt_codegen::core::provider::association;
+	using ecsact::rt_entt_codegen::core::provider::basic;
+	using ecsact::rt_entt_codegen::core::provider::lazy;
+	using ecsact::rt_entt_codegen::core::provider::notify;
 	using ecsact::rt_entt_codegen::system_util::is_notify_system;
-	using ecsact::rt_entt_codegen::system_util::print_system_notify_views;
-	using ecsact::rt_entt_codegen::util::method_printer;
 
-	auto sys_caps = ecsact::meta::system_capabilities(options.get_sys_like_id());
-	auto lazy_iteration_rate = 0;
+	assert(sys_like_id_variant != system_like_id_variant{});
 
-	auto exec_start_label_name =
-		std::format("exec_start_{}_", static_cast<int>(options.get_sys_like_id()));
+	auto lazy_provider = std::make_shared<lazy>(sys_like_id_variant);
+	auto association_provider =
+		std::make_shared<association>(sys_like_id_variant);
+	auto notify_provider = std::make_shared<notify>(sys_like_id_variant);
+	auto basic_provider = std::make_shared<basic>(sys_like_id_variant);
 
-	auto pending_lazy_exec_struct = std::format(
-		"::ecsact::entt::detail::pending_lazy_execution<::{}>",
-		options.system_name
-	);
+	std::vector<std::shared_ptr<system_provider>> system_providers{};
 
-	auto system_sorting_struct_name = std::format(
-		"::ecsact::entt::detail::system_sorted<{}>",
-		options.system_name
+	auto sys_details = ecsact_entt_system_details::from_system_like(
+		sys_like_id_variant.get_sys_like_id()
 	);
 
-	auto additional_view_components = std::vector<std::string>{};
+	int lazy_iteration_rate = 0;
 
-	if(options.is_system()) {
+	if(sys_like_id_variant.is_system()) {
 		lazy_iteration_rate = ecsact_meta_get_lazy_iteration_rate(
-			static_cast<ecsact_system_id>(options.get_sys_like_id())
+			static_cast<ecsact_system_id>(sys_like_id_variant.get_sys_like_id())
 		);
 	}
 
-	if(lazy_iteration_rate > 0) {
-		ctx.write(
-			"constexpr auto lazy_iteration_rate_ = ",
-			lazy_iteration_rate,
-			";\n\n"
-		);
-		ctx.write("auto iteration_count_ = 0;\n\n");
-		ctx.write(exec_start_label_name, ":\n");
-		additional_view_components.push_back(pending_lazy_exec_struct);
+	if(is_notify_system(sys_like_id_variant.get_sys_like_id())) {
+		system_providers.push_back(notify_provider);
```

Comment:
lets inline these providers. no need to construct them until this point since they're optional.
```suggestion
		system_providers.push_back(make_shared<lazy>(...));
```